pkg/kube/inject/inject.go

```diff
@@ -572,6 +572,29 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, typeMetadata *
 		"toLower":             strings.ToLower,
 	}
 
+	// Migrating from values.yaml to mesh or env (for istiod - set using fromEnv instead of values)
+	// This allows an optional config map with vendor-specific settings/overrides - alternative is creating
+	// vendor-specific injection templates or a huge mess.
+	funcMap["value"] = func(key string, def string) string {
```

Comment:
I don't get it. So in the template we have something like `{{ value "foo.bar" "" }}` and it will lookup the environment variable `foo_bar` if it exists, and if not it will get `.global.foo.bar`?

Why not just do something like `{{ valueOrDefault (getEnv "PILOT_FOO_BAR") .Values.global.foo.bar }}`? its much more explicit I think

Reply:
That will make other problems below go away

Reply:
We don't have getEnv - if I can add getEnv instead it's great - it was the first implementation :-)

But knowing people are sensitive to env variables - I tried to not expose them directly.

Reply:
Seems more desired to just explicitly add an getEnv function rather than to create one, mask it under a name `value` that does it anyways but is less flexible and doesn't work in a lot of cases?

Reply:
renamed to {{ env }}